lib/plugins/chat.js

```diff
@@ -108,43 +108,54 @@ function inject (bot, options) {
 
   addDefaultPatterns()
 
-  // Pre 1.19
-  bot._client.on('chat', (packet) => {
-    const msg = ChatMessage.fromNotch(packet.message)
-    const chatPositions = {
-      0: 'chat',
-      1: 'system',
-      2: 'game_info'
-    }
-    bot.emit('message', msg, chatPositions[packet.position], packet.sender, /* verified */ null)
-    bot.emit('messagestr', msg.toString(), chatPositions[packet.position], msg, packet.sender, /* verified */ null)
-    // Position 2 is the action bar
-    if (packet.position === 2) bot.emit('actionBar', msg, null)
-  })
+  bot._client.on('playerChat', (data) => {
+    const message = data.formattedMessage
+    const verified = data.verified
+    let msg
+    if (bot.supportFeature('clientsideChatFormatting')) {
+      let content
 
-  // 1.19+
-  bot._client.on('player_chat', (packet) => {
-    const message = packet.unsignedChatContent || packet.signedChatContent
-    let verified = false
-    const sender = bot.uuidToUsername[packet.senderUuid]
-    if (sender) {
-      const { profileKeys } = bot.players[sender]
-      if (profileKeys) verified = bot._client.verifyMessage(profileKeys.publicKey, packet)
-    }
-    const parameters = {
-      sender: JSON.parse(packet.senderName),
-      content: JSON.parse(message)
+      if (message) content = JSON.parse(message)
+      else if (options.allowInsecureChat && data.unsignedContent) content = JSON.parse(data.unsignedContent)
```

Comment:
> options.allowInsecureChat

Remove per prior comment

Reply:
Without this check, the behaviour of chat patterns (due to relying on the messagestr event) will be unpredictable as you can't compare to Vanilla.

Reply:
Removing for now. I still think we should keep it, but let's see what happens.